libraries/chain/combined_database.cpp

```diff
@@ -494,19 +514,16 @@ namespace eosio { namespace chain {
                         move_to_rocks = &row;
                      });
                      if (move_to_rocks) {
-                        auto buffer = std::vector<char>{};
-                        buffer.reserve(sizeof(uint64_t) + prefix_key.size() + move_to_rocks->kv_key.size());
-                        buffer.insert(std::end(buffer), prefix_key.data(), prefix_key.data() + prefix_key.size());
-                        b1::chain_kv::append_key(buffer, move_to_rocks->contract.to_uint64_t());
-                        buffer.insert(std::end(buffer), move_to_rocks->kv_key.data(),
-                                      move_to_rocks->kv_key.data() + move_to_rocks->kv_key.size());
+                        auto key = eosio::session::shared_bytes{sizeof(uint64_t) + prefix_key.size() + move_to_rocks->kv_key.size()};
+                        std::copy(std::begin(prefix_key), std::end(prefix_key), std::begin(key));
+                        b1::chain_kv::insert_key(key, prefix_key.size(), move_to_rocks->contract.to_uint64_t());
+                        std::copy(std::begin(move_to_rocks->kv_key), std::end(move_to_rocks->kv_key), std::begin(key) + prefix_key.size() + sizeof(uint64_t));
```

Comment:
So how is it safe for eosio::session::shared_bytes to share a key buffer that can be updated?  I thought the point was that it was created and was constant from then on.

Also, is this faster than using the make_shared_bytes method?

Reply:
That's how I originally intended it to be but it seemed like there was a lot of temporary buffers being created when I could just allow the user to write directly into the shared_bytes.  

It depends.  If you already have a buffer that you are wanting to copy from then you'd want to instantiate the shared_bytes with that buffer.  If you have to construct a buffer, then it will be a bit more efficient to just create the shared_bytes instance with a length and construct it in place.

Reply:
Discussion about this was on slack.